src/tests/ftest/object/same_key_different_value.py

```diff
@@ -0,0 +1,162 @@
+        Case1: Insert akey,dkey with single value
+               Insert same akey,dkey with array value
+               Result: should return -1001 Error.
+        Case2: Insert akey,dkey with single value
+               Punch the keys
+               Trigger aggregation
+               Insert same akey,dkey under same object with array value
+               Result: Should work without errors.
+        :avocado: tags=object,samekeydifferentvalue,vm,small
+        """
+
+        try:
+            # define akey,dkey, single value data and array value data
+            single_value_data = "a string that I want to stuff into an object"
+            array_value_data = []
+            array_value_data.append("data string one")
+            array_value_data.append("data string two")
+            array_value_data.append("data string tre")
+
+            dkey = "this is the dkey"
+            akey = "this is the akey"
+
+            # create an object and write single value data into it
+            obj, txn = self.container.write_an_obj(single_value_data,
+                                                   len(single_value_data)+1,
+                                                   dkey, akey, obj_cls=1)
+
+            # read the data back and make sure its correct
+            read_back_data = self.container.read_an_obj(
+                len(single_value_data)+1, dkey, akey, obj, txn)
+            if single_value_data != read_back_data.value:
+                print("data I wrote:" + single_value_data)
+                print("data I read back" + read_back_data.value)
+                self.fail("Write data, read it back, didn't match\n")
+
+            # write array value data to same keys, expected to fail
+            self.container.write_an_array_value(array_value_data,
+                                                dkey, akey, obj,
+                                                obj_cls=1)
+
+            # above line is expected to return an error, if not fail the test
+            self.fail("Array value write to existing single value key"
+                      + " should have failed\n")
+        except DaosApiError as excp:
+            if "-1001" not in str(excp):
+                print(excp)
+                self.fail("Should have failed with -1001 error message,"
+                          + " but it did not\n")
+        try:
+            # punch akey and dkey
+            obj.punch_akeys(0, dkey, [akey])
+            obj.punch_dkeys(0, [dkey])
+
+            # trigger aggregation
+            self.container.aggregate(self.container.coh, 0)
+
+            # write array value to same set of keys under same object
+            # it should be successful this time
+            obj, txn = self.container.write_an_array_value(array_value_data,
+                                                           dkey, akey, obj,
+                                                           obj_cls=1)
+        # test should fail if exception occurs
```

Comment:
the result of this write call could be either success, or NO_PERM, since aggregation could have occurred or it might not have been done yet when the write is issued. So i would add -1001 as a valid result in addition to success.
See the comments in the ticket.

Reply:
Updated test takes care of this.
Also one more test case was added with no aggregation in the middle and then trying to write an array value type. Ticket has been modified for detailed reference.